engine/crates/integration-tests/src/federation/builder/test_runtime.rs

```diff
@@ -1,8 +1,8 @@
 use grafbase_telemetry::{metrics, otel::opentelemetry};
 use runtime::{entity_cache::EntityCache, hooks::DynamicHooks, trusted_documents_client};
 use runtime_local::{
-    rate_limiting::in_memory::key_based::InMemoryRateLimiter, InMemoryEntityCache, InMemoryHotCacheFactory,
-    InMemoryKvStore, NativeFetcher,
+    rate_limiting::in_memory::key_based::InMemoryRateLimiter, InMemoryEntityCache, InMemoryKvStore,
```

Comment:
:bulb: **Performance**
The addition of `InMemoryOperationCacheFactory` as a type and its associated method `operation_cache_factory` could potentially lead to performance issues if the cache is not managed efficiently. In-memory caches can consume significant memory resources, especially if the cache size is not controlled or if it holds large objects. Consider implementing a cache eviction policy, such as LRU (Least Recently Used), to manage the memory footprint effectively.
```suggestion
Implement a cache eviction policy within the `InMemoryOperationCacheFactory` to ensure that the cache does not grow indefinitely. This can be done by using a data structure that supports efficient eviction, such as an LRU cache.
```